library/qtest/src/Functions.qs

```diff
@@ -47,6 +47,26 @@ function RunAllTestCases<'T : Eq + Show>(test_cases : (String, () -> 'T, 'T)[])
     Mapped((name, case, result) -> TestCase(name, case, result), test_cases)
 }
 
+/// # Summary
+/// Given a function to test and an array of test cases of the form (input, expected_output), and a test mode, runs the test cases and returns the result of the test mode.
+///
+/// # Inputs
+/// - `test_suite_name` : A string representing the name of the test suite.
+/// - `func` : The function to test.
+/// - `test_cases` : An array of tuples of the form (input, expected_output).
+/// - `mode` : A function that takes an array of tuples of the form (test_name, test_case, expected_output) and returns a value of type 'U.
+///            Intended to be either `Qtest.Functions.CheckAllTestCases` or `Qtest.Functions.RunAllTestCases`.
+///
+/// # Example
+/// ```qsharp
+/// TestMatrix("Add One", x -> x + 1 [(2, 3), (3, 4)], CheckAllTestCases);
+/// ```
+
+function TestMatrix<'T, 'O : Show + Eq, 'U>(test_suite_name : String, func : 'T -> 'O, test_cases : ('T, 'O)[], mode : ((String, () -> 'O, 'O)[]) -> 'U) : 'U {
```

Comment:
Thanks @sezna for taking this up. Nice addition of `mode` function for further flexibility :-)

**Minor nitpick:** RunTestMatrix, thoughts?

Reply:
I like the mode function. We can leave it as it is if you think renaming the function causes confusion. I suggested RunTestMatrix since it gives the feeling that we are doing something. TestMatrix gives the data vibe. I thought mode=CheckAllTestCases/RunAllTestCases in RunTestMatrix is good enough.

PS: sent from mobile.not responsible for autocorrect typos.

Reply:
I kind of like just having `TestMatrix`, due to the preexisting Check/Run dichotomy. Do you feel strongly about this?